index.js

```diff
@@ -56,12 +56,12 @@ function interpolate(t, degree, points, knots, weights, result) {
     v[i][d] = weights[i];
   }
 
-  // l (level) goes from 1 to the curve degree
+  // l (level) goes from 1 to the curve degree + 1
   var alpha;
-  for(l=1; l<=degree; l++) {
+  for(l=1; l<=degree+1; l++) {
     // build level l of the pyramid
-    for(i=s; i>s-degree+l; i--) {
-      alpha = (t - knots[i]) / (knots[i+degree-l] - knots[i]);
+    for(i=s; i>s-degree-1+l; i--) {
```

Comment:
hm, the `knots.length` relation seems relatively clear from the code already due to the length check, with Error if it's the wrong length, only thing I would change there is, rather than saying that the length is "bad", say that the length is not the expected "number of control points plus spline order", and maybe capture the value `degree + 1` as the value `order` in the code so that it's a traceable named value (which a good minifier can optimize back out of existence).

As for the level indicator, I kind of like seeing a first step (correctly) called step `1` rather than step `0`. Sure, array offsets start at 0 due to pointer logic, but maths algorithms are not constrained by pointer logic, and arrays in JS aren't true arrays in the C sense anyway (they're still the same key/object pair containers as regular objects, just with iteration logic baked in on top) so missing out on that spot `0` doesn't actually have any impact.

A long-hand form of the algorithm might be nice, but a good intermediate step might be to just unroll the variable naming, so that `l` becomes `level`, `s` becomes `segment`, and `v` gets a name that makes it obvious what `v[i][j]` is when compared to "most" literature around De Boor's algorithm, like calling it `N` and then using indices `i` and `k` to match the many (most? it's hard to tell in this case) explanations in literature.